src/utils/barcode.ts

```
@@ -0,0 +1,81 @@
+import * as bwipjs from "bwip-js";
+import { tryCatch } from "fp-ts/lib/TaskEither";
+import * as JsBarcode from "jsbarcode";
+import { DOMImplementation, XMLSerializer } from "xmldom";
+
+export interface IBarcodeOutput {
+  png: string;
+  pngMimeType: string;
+  svg: string;
+  svgMimeType: string;
+}
+
+interface IBwipOptions {
+  bcid: string;
+  text: string;
+  includetext: boolean;
+}
+
+const toBase64Png = (options: IBwipOptions) => {
+  return tryCatch(
+    () => {
+      return new Promise<string>((resolve, reject) =>
+        bwipjs.toBuffer(options, (err, png) => {
+          if (err == null) {
+            return resolve(png.toString("base64"));
+          } else {
+            return reject(new Error("Cannot generate png barcode"));
```

probably better to pass `err` along instead of swallowing it?